src/shogun/base/SGObject.cpp

```diff
@@ -70,7 +70,14 @@ namespace shogun
 			{
 				error("Can not register {} twice", tag.name().c_str());
 			}
-			map[tag] = parameter;
+			map.emplace_back(tag, parameter);
+		}
+
+		auto find(const BaseTag& tag) const
+		{
+			for (auto it = map.cbegin(); it != map.cend(); it++)
+				if (it->first == tag)
+					return it;
```

Comment:
shouldn't this return map.cend() after the loop?

Reply:
mostdef... this was f*!@# up during cleanup before commit